fuse/nodefs/fsconnector.go

```diff
@@ -126,9 +135,23 @@ func (c *FileSystemConnector) forgetUpdate(nodeID uint64, forgetCount int) {
 
 	if forgotten, handled := c.inodeMap.Forget(nodeID, forgetCount); forgotten {
 		node := (*Inode)(unsafe.Pointer(handled))
+		if node.IsDir() && len(node.Children()) > 0 {
+			// This directory node has children that have not yet been forgotten.
+			// We don't want to remove it from the tree (i.e. remove it from its parent)
+			// because the children may have parent pointers to it.
+			// A later LOOKUP will return this node with a new NodeID.
+			return
+		}
 		node.mount.treeLock.Lock()
-		c.recursiveConsiderDropInode(node)
+		// Remove Inode from all parents
```

Comment:
If we have the full tree in nodefs (actually, with hard links, it's not a tree anymore, it's DAG I believe), then pathfs should drop it's own Parent pointers to avoid duplication.